### PR TITLE
BUG: Reserved keywords

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # https://github.com/pysat/pysatMissions/issues/50

Adds a list of reserved keywords to documentation. I also added a test if a reserved keyword (intended for instrument support function) is provided. Plus, a unit test for reserved keywords.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

Added new unit test

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

**Review by jklenzing — APPROVED** [review]
Looks good.  Verified that I get the error message over at pysatMissions.  Thanks @rstoneback!

File: pysat/_instrument.py [hunk 1]
```diff
@@ -3785,6 +3793,20 @@ def _kwargs_keys_to_func_name(kwargs_key):
...
+
+    Returns
+    -------
+    out_list : list
+       List of strings for reserved keywords
+
+    """
+
+    return ['fnames', 'inst_id', 'tag', 'date_array', 'data_path',
+            'format_str', 'supported_tags', 'start', 'stop', 'freq']
```
Comment (aburrell) [inline pysat/_instrument.py:3807]: Would it not be simpler to just define this as a hidden attribute?

**Review by aburrell — COMMENTED** [review]
An implementation question.

File: pysat/_instrument.py (hunk 1 above)
Comment (rstoneback) [reply pysat/_instrument.py]: I defined it as a hidden attribute at _instrument.py level. A copy is attached to each Instrument object as `inst.kwargs_reserved` to go along with `inst.kwargs` and `inst.kwargs_supported`

**Comment by rstoneback** [comment]
Failing for unrelated reasons:
```
ValueError: passing 'axis' to Dataset reduce methods is ambiguous. Please use 'dim' instead.
```

**Review by jklenzing — APPROVED** [review]
Looks good!

**Pull request merged**